processing/src/main/java/io/druid/segment/StringDimensionHandler.java

```diff
@@ -120,24 +120,20 @@ public void validateSortedEncodedArrays(
               dimValName
           );
         }
+      } else {
```

Comment:
@jon-wei thanks, I see now. Semantically it is ok, but the fact that the loop below carefully accounts for the situation when `lhs.length` != `rhs.length`, while now they are always equal, is a sign that the original author of this code had something different in mind. Probably he initially wanted to allow `lhs.length = 2 and rhs.length = 0` case to pass verification.

So I think that the loop below should be simplified now.